### PR TITLE
Feature shapefile upload

### PR DESCRIPTION
**Pull request by ikas**
### Overview

This PR adds a new endpoint `POST /api/v1/projects/:id/features/shapefile` that allows users to upload feature shapefiles.

To use this endpoint, you'll need to provide a `.zip` file containing the different format files that compose a shapefile, and send it as `multipart/form-data` to the endpoint specified above.

The upload happens synchronously, so once you receive the response, the data should already be available in the database. Also, keep in mind there's a limit on the maximum size of the uploaded file, currently set to 100MB (this limit is configurable via env vars).

### Designs

[Figma](https://www.figma.com/file/hq0BZNB9fzyFSbEUgQIHdK/Marxan-Visual_V02?node-id=2991%3A2492)

_This PR does not contain the implementation of the UI for this feature, just the BE side of things._

### Testing instructions

Use the new endpoint and check that data gets correctly imported to the databases (API and geo) - check the JIRA ticket below for more details on where the data should be placed.

### Feature relevant tickets

[MARXAN-543](https://vizzuality.atlassian.net/browse/MARXAN-543)

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: api/apps/api/src/modules/geo-features/geo-feature.api.entity.ts
Comment (ikas) [inline api/apps/api/src/modules/geo-features/geo-feature.api.entity.ts:22]: had to make this to be able to access the `id` of the entity after inserting it into the DB

File: api/apps/api/src/modules/geo-features/geo-features.service.ts
Comment (ikas) [inline api/apps/api/src/modules/geo-features/geo-features.service.ts:273]: transactions here are little bit sketchy, since I'm writing data to 2 separate DBs :thinking: if you think of a better solution let me know :smile:

File: api/apps/api/src/modules/projects/projects.controller.ts
Comment (ikas) [inline api/apps/api/src/modules/projects/projects.controller.ts:264]: fixing this right now

File: api/apps/api/src/modules/geo-features/geo-features.service.ts
Comment (hotzevzl) [reply api/apps/api/src/modules/geo-features/geo-features.service.ts:273]: yes, it occurred to me with a huge 🤦🏼 when I saw your PR while trying to help with the makevalid bit - I think what you do here will be fine in most circumstances. if we do run into issues in a substantive way I'd think about a final check both in case of success or failure, but since I'm thinking about a grim reaper/garbage collection feature that we will need in any case to be able to invalidate stuff when users edit settings earlier on in the workflow, this - if needed at all - could be done when we implement that.

File: api/apps/api/src/modules/projects/projects.controller.ts
Comment (ikas) [reply api/apps/api/src/modules/projects/projects.controller.ts]: fixed!

**Review by hotzevzl — APPROVED** [review]
Thanks @ikas! Epic PR, thanks so much for your dedication this week 🙇🏼 

I'd ask @aagm to please cross check that there are no blind spots (which, if any, would mainly be due to blind spots in my specification, tbh) before merging, though I'm itching to merge RIGHT NOW so that I can use this feature via bots I'm working on 😅

**Comment by hotzevzl** [comment]
oh well I hadn't notice the test failures - though they all seem to be identical and only due to an import lookup issue, so this should be easy to fix (if you get stuck, probably Dominik can help fixing this while blindfolded)

File: api/apps/api/src/modules/projects/projects.controller.ts
Comment (Dyostiq) [inline api/apps/api/src/modules/projects/projects.controller.ts:76]: It should not be used here.

It's a service from the geo app, and we should not share anything between the app `as is` but by shared libs.
The second thing is the service uses mapshaper under the hood, and we wanted to keep heavy calculation on the geo side – with this, we're breaking the rule.

File: api/apps/api/src/modules/geo-features/geo-feature.api.entity.ts
Comment (kgajowy) [inline api/apps/api/src/modules/geo-features/geo-feature.api.entity.ts:61]: Why is this field necessary? Actual status usually come from ApiEvents (and in this case, should rather be related to some Scenario* event)

File: api/apps/api/src/modules/geo-features/geo-feature.geo.entity.ts
Comment (kgajowy) [inline api/apps/api/src/modules/geo-features/geo-feature.geo.entity.ts:33]: Are we able to narrow the type?

File: api/apps/api/src/modules/geo-features/geo-feature.geo.entity.ts
Comment (kgajowy) [inline api/apps/api/src/modules/geo-features/geo-feature.geo.entity.ts:36]: It will cause issues when generating migrations. 
```
@Column(`enum`, { enum: SourceType, nullable: strue })
source?: SourceType | null
```

File: api/apps/api/src/modules/geo-features/geo-features.service.ts
Comment (kgajowy) [inline api/apps/api/src/modules/geo-features/geo-features.service.ts:291]: [minor] not sure if iterating gives us any profit there (could be just simple `const apiDbConnection = getConnection(DbConnections.Api...)`

File: api/apps/api/src/modules/geo-features/geo-features.service.ts
Comment (kgajowy) [inline api/apps/api/src/modules/geo-features/geo-features.service.ts:318]: How about doing those in `bulk`?

File: api/apps/api/src/modules/geo-features/geo-feature.api.entity.ts
Comment (hotzevzl) [reply api/apps/api/src/modules/geo-features/geo-feature.api.entity.ts:61]: I think this was done to keep to the current table structure, but we could probably even drop this column and related entity property, especially if doing the insert synchronously as in this initial implementation.

File: api/apps/api/src/modules/geo-features/geo-feature.geo.entity.ts
Comment (hotzevzl) [reply api/apps/api/src/modules/geo-features/geo-feature.geo.entity.ts:33]: are you asking if we can further narrow it, or if the narrowing here is accurate?

if the latter, looking at the dbase IV spec, I think we may only be missing boolean. I advised on this specific typing because of the way we import features via ETL pipelines (where bools are imported as string), but it won't hurt to add boolean here if that's what you meant.

Comment (kgajowy) [reply api/apps/api/src/modules/geo-features/geo-feature.geo.entity.ts:33]: @hotzevzl I meant to provide more detailed structure rather than generic "everything can be there"

**Pull request merged**
